File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        f = dict({})
        for n in nums:
            if f.get(n) is not None:
                f[n] += 1
            else:
                f[n] = 1
        for k in f.keys():
            if f[k] > 1:
                return True
        return False

File: Data Structures & Algorithms/three-integer-sum/submission-2.java
class Solution {

    public List<List<Integer>> threeSum(int[] nums) {
        if(nums.length == 0){
            return new ArrayList<>();
        }
        Set<List<Integer>> res = new HashSet<>(); 
        List<Integer> numL = Arrays.stream(nums).boxed().collect(Collectors.toList());
        Collections.sort(numL);
        for(int i = 0; i < numL.size(); i++){
            int target = -numL.get(i);
            int l = i + 1;
            int r = numL.size() - 1;
            while(l < r){
                if(numL.get(l) + numL.get(r) == target){
                    res.add(Arrays.asList(numL.get(i), numL.get(l), numL.get(r)));
                }
                if(numL.get(l) + numL.get(r) > target){
                    r--;
                }else{
                    l++;
                }
            }         
        }
        return new ArrayList<>(res);
    }
}
